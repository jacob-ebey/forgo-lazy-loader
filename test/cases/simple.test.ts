import childProcess from "child_process";
import path from "path";

import request from "supertest";

describe("simple", () => {
  it("works", async () => {
    await new Promise<void>((resolve, reject) => {
      childProcess.exec(
        "yarn",
        { cwd: path.join(__dirname, "simple") },
        (err, stdout, stderr) => {
          if (err) {
            return reject(err);
          }
          console.log(stderr);
          console.log(stdout);
          return resolve();
        }
      );
    });

    await new Promise<void>((resolve, reject) => {
      childProcess.exec(
        "yarn build",
        { cwd: path.join(__dirname, "simple") },
        (err, stdout, stderr) => {
          if (err) {
            return reject(err);
          }
          console.log(stderr);
          console.log(stdout);
          return resolve();
        }
      );
    });

    const app = require("./simple/dist/server/main");
    const res = await request(app.default).get("/").expect(200);
    expect(res.text).toMatchSnapshot();
  });
});
