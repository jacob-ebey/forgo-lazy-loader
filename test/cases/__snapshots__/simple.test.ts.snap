// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple works 1`] = `
"<!DOCTYPE html>
<html><head><link href=\\"/static/P7CO8vTQqtFIUzF90z+rfRBG2SA.css\\" rel=\\"preload\\" as=\\"style\\"><link href=\\"/static/P7CO8vTQqtFIUzF90z+rfRBG2SA.js\\" rel=\\"preload\\" as=\\"script\\"><link href=\\"/static/main.js\\" rel=\\"preload\\" as=\\"script\\"><link href=\\"/static/P7CO8vTQqtFIUzF90z+rfRBG2SA.css\\" type=\\"text/css\\" rel=\\"stylesheet\\"></head><body><div id=\\"__forgo\\"><div><main><h1 class=\\"XGRxiOrAtt9rIMaoJ9R4W\\">I&#39;m Split! :D and have styles</h1><button>Log to the console</button></main></div></div><script src=\\"/static/main.js\\"></script></body></html>"
`;
